### DISCUSSION TITLE
Feature Request: IN Operator multi column support

### DISCUSSION THREAD

**Discussion by moutella** [duckdb/duckdb#11301] [Ideas]
Currently, IN operator only works with subqueries with one column, as stated [here](https://duckdb.org/docs/sql/expressions/in.html)

I am starting to implement duckdb to speed up development of a project while the database migrations haven't been applied to the servers, and while testing out my queries, I found out that I couldn't do 
```sql
select * from table where ("field1", "field2") in (select "field1", "field2" from table)
```

To work around it, all I had to do was "CONCAT" my fields, and it worked
```sql
select * from table where concat("field1", "field2") in (select concat("field1", "field2") from table)
```

I am not an expert in databases, but I believe that using "CONCAT" will make my queries slower than they should be. So I was wondering if, instead of using "CONCAT" on my query, duckdb could be able to apply the "CONCAT" under the rugs, just to make the IN operator compatible with multiple columns.

I tried looking around the source code and felt like it's definitely way out of my league to implement this, so I decided to start a discussion. All I found out is that I think it's around [here](https://github.com/duckdb/duckdb/blob/2f58548f84103e18ea9ccfcab28de75714fecddd/src/planner/binder/expression/bind_subquery_expression.cpp#L57C3-L59C4)

**Comment by cmdlineluser** [comment]
You should be able to use a [struct](https://duckdb.org/docs/sql/data_types/struct.html) instead of concat:

```sql
select ("field1", "field2") 
-- same as: select row("field1", "field2")
```

**Reply by moutella** [reply]
Thanks! Changed my query here and it worked like that :)
